lib/course/middleware/assets.rb

```diff
@@ -0,0 +1,59 @@
+module Course
+  module Middleware
+    class Assets
+      attr_reader :app
+
+      REQUEST_METHOD_GET = "GET".freeze
+      PUBLIC_URL = "/public".freeze
+
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        if get?(env) && public_url?(env)
+          handle_public_request(env)
+        else
+          app.call(env)
+        end
+      end
+
+      private
+
+      def get?(env)
+        env["REQUEST_METHOD"] == REQUEST_METHOD_GET
+      end
+
+      def public_url?(env)
+        env["REQUEST_PATH"].start_with?(PUBLIC_URL)
+      end
+
+      def handle_public_request(env)
+        return [404, { "content-type" => "text/plain" }, ["Source not found"]] if danger_path?(env)
```

Comment:
Think about it, it looks like you are duplicating logic from Unsuccess.
So if you reorder middlewares the other way... How it can be improved?

Reply:
Fixed, return simple 404 status